#### Load Necessary Libraries

In [16]:
from sklearn.model_selection import train_test_split
from sklearn.datasets import make_moons
import numpy as np

from sklearn.ensemble import VotingClassifier
from sklearn.linear_model import LogisticRegression
from sklearn.tree import DecisionTreeClassifier

from sklearn.metrics import accuracy_score

#### generate a dataset using make_moons from scikit-learn and then adding some random noise to it. After that,split the dataset into training and testing sets using train_test_split from scikit-learn.

In [17]:
X, y = make_moons(n_samples=100, noise=1)
X = X + (np.random.random(size=100 * 2).reshape((100, 2)) ** 2)
X_train, X_test, y_train, y_test = train_test_split(X, y)

#### two classifier objects are created: (log_clf) and (dt_clf)

In [18]:
log_clf = LogisticRegression()
dt_clf = DecisionTreeClassifier()

###  ensemble learning method in scikit-learn

#### combines the predictions from a logistic regression classifier (log_clf) and a decision tree classifier (dt_clf) using the majority vote (voting='hard') method.

In [19]:
voting_clf = VotingClassifier(
    estimators=[('lr', log_clf), ('dt', dt_clf)],
    voting='hard')

#### fits the VotingClassifier to the training data (X_train and y_train)

In [20]:
voting_clf.fit(X_train, y_train) 

VotingClassifier(estimators=[('lr', LogisticRegression()),
                             ('dt', DecisionTreeClassifier())])

#### predictions on the test data, and then prints the accuracy score for each classifier

In [21]:
for clf in (log_clf,dt_clf,voting_clf):
    clf.fit(X_train, y_train)
    y_pred = clf.predict(X_test)
    print(clf.__class__.__name__, accuracy_score(y_test, y_pred))

LogisticRegression 0.84
DecisionTreeClassifier 0.64
VotingClassifier 0.72


* Observation:
    - This would mean that the logistic regression classifier achieved an accuracy of approximately 84%, the decision tree classifier achieved an accuracy of approximately 64%, and the voting classifier achieved an accuracy of approximately 72% on the test data.



